### DISCUSSION TITLE
@stlite/desktop app - access local server

### DISCUSSION THREAD

**Discussion by ACR-GH** [whitphx/stlite#1043] [General]
Hi there.

I have a streamlit app that uses SQLAlchemy + python-tds to interact with a SQL Server database in the local network. 
This works fine on streamlit but the connection times out when using the Eletron app generated using @stlite/desktop

Any thoughts on what might be the reason and any workarounds?

**Comment by whitphx** [comment]
I don't know about `python-tds`, but the time-out error around the networking code is often from [Pyodide's network restriction](https://github.com/whitphx/stlite?tab=readme-ov-file#http-requests).

If you can inject a custom networking handler into `pyodide-tds`, this can be a solution, like [this httpx case](https://gist.github.com/florimondmanca/d56764d78d748eb9f73165da388e546e) where a custom networking layer using urllib3 that works on Pyodide is injected into `httpx`.
